### ISSUE TITLE
Hardware Decoding is broken for certain Flatpak Apps

### ISSUE THREAD

**Issue by ranger149** [ValveSoftware/SteamOS#1014]
#### Your system information

* Steam client version: 1680827821
* SteamOS version: 3.4.6
* Opted into Steam client beta?: Yes
* Opted into SteamOS beta?: Yes
* Have you checked for updates in Settings > System?: Yes

#### Please describe your issue in as much detail as possible:

After many users updated their flatpaks via the discover store, It appears Hardware Decoding has been broken for Flatpak apps. E.X. Moonlight and Chiaki. I can confirm, however, using the AppImage version of Moonlight does not reproduce this issue. Also this issue occurs regardless of Steam Beta, Preview, Or Stable OS versions. I'm not sure if this is an issue specifically with SteamOS, or an issue with an update pushed via the Discover store, but it's effecting many Steam Deck users all the same.

For reference that this issue is widespread:

https://github.com/streetpea/chiaki4deck/issues/63
https://www.reddit.com/r/SteamDeck/comments/12pu215/problem_with_moonlight_and_sunshine_decoder/
https://www.reddit.com/r/SteamDeck/comments/12qqwod/can_anybody_help_me_pls_i_can_not_remote_play/
https://www.reddit.com/r/SteamDeck/comments/12qfq7j/steamos_346_chiaki_error/

#### Steps for reproducing this issue:

1.  Enable hardware decoding on Moonlight and/or Chiaki
2.  Launch Moonlight or Chiaki and you'll be greeted with an error about hardware decoding.

**Comment by waspennator** [comment]
If it still works via appimages, then I'm guessing the fault lies with whichever discovery store update broke hardware decoding for flatpaks

**Comment by kisak-valve** [comment]
Hello @ranger149, the mesa video driver stack including the va-api implementation used inside of Flatpak containers is exclusively provided by the org.freedesktop.Platform runtime environment. This isn't modified or maintained by SteamOS.

Report this issue to Flatpak's package maintainer(s) for mesa.

Sorry, but this isn't a SteamOS issue unless there's a stronger hint to ponder.

**Comment by lostgoat** [comment]
This is the actual error from ffmpeg in the container:
```
[AVHWDeviceContext @ 0x5591d91fe080] libva: /usr/lib/x86_64-linux-gnu/GL/lib/dri/radeonsi_drv_video.so has no function __vaDriverInit_1_0
[AVHWDeviceContext @ 0x5591d91fe080] Failed to initialise VAAPI connection: -1 (unknown libva error).
Device creation failed: -5.
[h264 @ 0x5591d91d7f40] No device available for decoder: device type vaapi needed for codec h264.
Stream mapping:
  Stream #0:0 -> #0:0 (h264 (native) -> mpeg4 (native))
  Stream #0:1 -> #0:1 (aac (libfdk_aac) -> mp3 (libmp3lame))
Device setup failed for decoder on input stream #0:0 : Input/output error
```

**Comment by kisak-valve** [comment]
Possibly related to https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1528#note_1323212034.

**Comment by lostgoat** [comment]
I posted some details on the problem here:
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1579

TLDR: it looks like the org.kde runtime provides an outdated libva that is incompatible with the new mesa in flatpak.

**Comment by ranger149** [comment]
Thank you for the help finding the issue!

**Comment by helliongame** [comment]
Any news about this issue?

**Comment by lostgoat** [comment]
`org.kde.Platform` version `5.15-22.08` and `6.4` have been updated with the new `libva`. This should fix hardware acceleration for Chiaki/Moonlight/OBS.

**Comment by lostgoat** [comment]
Please update your flatpaks and retest and let me know if everything is in working order for you.

**Comment by ranger149** [comment]
I'm able to confirm after updating the KDE platform the issue was resolved. Thank you!

**Comment by ranger149** [comment]
Issue is resolved. Flatpaks are now able to use hardware decoding after the KDE runtime was rebuilt. Closed as not planned as this issue was not with SteamOS, and this was only open for tracking purposes.

**Comment by lostgoat** [comment]
Thanks for the report!

Even though this didn't turn out to be a SteamOS issue directly, reports like these are still useful to have in this tracker so we are aware of the problem. If this happens again in the future (hopefully not with the new libva process for flatpak!) feel free to open an new issue here.

**Comment by waspennator** [comment]
Apparently someone I'd saying the flatpak version of moonlight is still broken
